ipv6/miredo/Makefile

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (C) 2006-2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+
+# Developer notes:
+#   - Make sure you have gawk, othwerise you get weird syntax errors about "("
+#     and ")".
+#   - Install ccache, otherwise you'll get configure error:
+#     "C compiler cannot create executables"
+#   - Shared libraries are a pain, so we just compile the binary static.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=miredo
+PKG_VERSION:=1.2.6
+PKG_RELEASE:=1
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/miredo-$(PKG_VERSION)
+PKG_SOURCE:=miredo-1.2.6.tar.xz
+PKG_SOURCE_URL:=https://www.remlab.net/files/miredo/
+PKG_SHA256SUM:=fa26d2f4a405415833669e2e2e22677b225d8f83600844645d5683535ea43149
+PKG_CAT:=xzcat
+PKG_LICENSE:=GPL-2.0
+PKG_LICENSE_FILES:=COPYING
+PKG_MAINTAINER:=Petr Pudlak <petr.mvd@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/miredo
+	SECTION:=net
+	CATEGORY:=Network
+	TITLE:=Teredo IPv6 tunneling utility
+	URL:=https://www.remlab.net/miredo/
+	DEPENDS:=+libpthread +librt +ip +kmod-ipv6 +kmod-tun
+endef
+
+define Package/miredo/description
+ Miredo is an open-source Teredo IPv6 tunneling software, for Linux and the BSD
+ operating systems. It includes functional implementations of all components of
+ the Teredo specification (client, relay and server). It is meant to provide
+ IPv6 connectivity even from behind NAT devices.
+endef
+
+CONFIGURE_VARS += CFLAGS="$(TARGET_CFLAGS) -std=gnu99"
+
+define Build/Configure
```

Comment:
Works, thank you.

I was struggling a bit with shared libraries as `libtool` creates wrappers instead of the actual programs. But pulling the original ones from its helper directories like src/.libs/miredo works, I updated the patch to do this. (Maybe there's a better way, I'll keep looking.) However, when using shared libraries, the package is actually larger than when using static builds. In that case, is it still worth using shared, considering that it's very unlikely another program will ever use the libraries?